src/main/java/net/smartcosmos/dao/relationships/util/RelationshipPersistenceUtil.java

```diff
@@ -0,0 +1,119 @@
+package net.smartcosmos.dao.relationships.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.data.domain.Sort;
+
+import net.smartcosmos.dao.relationships.domain.RelationshipEntity;
+import net.smartcosmos.dao.relationships.SortOrder;
+import net.smartcosmos.dao.relationships.domain.RelationshipEntity;
+import net.smartcosmos.dto.relationships.Page;
+import net.smartcosmos.dto.relationships.RelationshipResponse;
+
+public class RelationshipPersistenceUtil {
+
+
+    /**
+     * Transforms a field name for a sorted query to a valid case-sensitive field name that exists in the entity class.
+     * Returns the input field name, if it does not exist in the entity class.
+     *
+     * @param fieldName the input field name
+     * @return the case-corrected field name
+     */
+    public static String normalizeFieldName(String fieldName) {
```

Comment:
you can short-circuit a null or empty string `""` as the first test then bad coding does not cause worst case performance